### ISSUE TITLE
Linking text annotations via relationships: UI prototype

### ISSUE THREAD

**Issue by rsimon** [pelagios/recogito2#497]

**Comment by rsimon** [comment]
Stage 1: drawing connections.

- get them working correctly in all four quadrants
- get them working correctly in above/below configurations (notch should attach at lower/upper end of the annotation SPAN accordingly
- make sure everything works when scrolling before/after/while drawing the connection
- allow users to click once-move-click again as well as hold-and-drag
- proper notch placement for overlapping annotations
- optional: vertical snap to line when moving the mouse

**Comment by rsimon** [comment]
Stage 2: editing tags (we'll keep it to tags for the time being)

**Comment by rsimon** [comment]
__Stage 3__: save/restore

**Comment by rsimon** [comment]
Pasting an animated design preview below:

![ezgif-1-317464bb74](https://user-images.githubusercontent.com/470971/39625831-9bfd9656-4f9f-11e8-825a-d9538f554371.gif)

**Comment by eltonteb** [comment]
Great demo! An initial question: where are the categories (motherOf, fatherOf) coming from? Are these categories predetermined or are they produced by the user? Their precise form reminded me of the kind of regular expressions you'd see in coding: I'm not sure whether I would have described the relations in exactly this way, but I can see that it would be helpful to for consistency and more general applicability (reuse).

**Comment by rsimon** [comment]
Nope, they’re just freeform tags. Like the ones you use in the annotations. I.e. no limitations or predetermined structure whatsoever. Meanwhile, I also added the same dropdown autocomplete functionality, so it’s easier to tag consistently.
